### ISSUE TITLE
Keep file structure when using exportsOverride

### ISSUE THREAD

**Issue by eskimoblood** [yatskevich/grunt-bower-task#21]
I load a requireJs based lib. As all files that mentioned in the exportsOverride block are put at the main level of `lib/js/myDependency/` I cant build the lib with requireJs as the require paths are broken now.

**Comment by fantactuka** [comment]
I guess you can use `layout` option that could be a function, that responsible for strategy of files copying. See docs here https://github.com/yatskevich/grunt-bower-task#optionslayout

**Comment by eskimoblood** [comment]
Sure, but wouldn't it be the expected behavior that the file structure keeps the same  as in the original repo instead of flatting all files in the same folder.

**Comment by fantactuka** [comment]
Can you bring in an example of exportsOverride you have?

**Comment by eskimoblood** [comment]
Ok, I have problem with this repo: https://github.com/rsolomon/JSV. It has a nested file structure and us requirejs. So when I load this grunt-bower all files put in the same folder and I have to move them afterwards in the right folder again. This will be a problem for every requirejs based repo.

**Comment by fantactuka** [comment]
Yes, now I see the problem. I guess this is something that was not considered in API. Good point for enhancement

**Comment by hamishtaplin** [comment]
I would like to see this, too.

**Comment by DoCode** [comment]
+1

**Comment by ghost** [comment]
+1

**Comment by RishiRamraj** [comment]
+1

**Comment by cvn** [comment]
+1

**Comment by kenwarner** [comment]
+1

**Comment by dr-potato** [comment]
+1

**Comment by cvrebert** [comment]
Yes! Please stop messing with packages' layouts!
AFAIK, there's basically no good way for packages to support having their directory structure reorganized by your tool. References to images+fonts within Less/CSS/Sass break, and then users often blame the package instead of grunt-bower-task.

**Comment by dr-potato** [comment]
In the meantime, for people trying to make this work with their grunt setup: [grunt-bower](https://github.com/curist/grunt-bower) does seem to support keeping the folder structure intact. Only thing it doesn't do is install bower packages, but [grunt-exec](https://github.com/jharding/grunt-exec) could take care of that.  

At this point it might make more sense to just use grunt-contrib-copy.

**Comment by kevin-smets** [comment]
Any news on this one?

**Comment by jeroenpelgrims** [comment]
+1

**Comment by jcano** [comment]
+1 This is very important indeed

**Comment by MikeMengell** [comment]
+1

**Comment by j3ffb** [comment]
+1

**Comment by afloyd** [comment]
+1

**Comment by andreicek** [comment]
Obnoxious comment but, +1

**Comment by black78** [comment]
+1 it need for copy boost less files structure. Please.
